src/mlpack/tests/ksinit_test.cpp

```diff
@@ -0,0 +1,315 @@
+    
+    validationTestLabels = 
+        trainLabels.submat(0, i, trainLabels.n_rows - 1, 
+                           i + validationDataSize - 1);
+
+    validationTrainData = trainData;
+    validationTrainData.shed_cols(i, i + validationDataSize - 1);
+    
+
+    validationTrainLabels = trainLabels;
+    validationTrainLabels.shed_cols(i, i + validationDataSize - 1);
+    
+    double trainError, validationError;
+
+    BuildVanillaNetwork(validationTrainData, validationTrainLabels, 
+         validationTestData, validationTestLabels, hiddenLayerSize, maxEpochs, 
+         validationTrainLabels.n_rows, trainError, validationError);
+
+    trainErrorAvg += trainError;
+    validationErrorAvg += validationError;
+
+  }
+
+  trainErrorAvg /= k;
+  validationErrorAvg  /= k;
+
+  Error e;
+
+  e.trainError = trainErrorAvg;
+  e.validationError = validationErrorAvg;
+
+  return e;
+
+}
+
+/*
+  AvgCrossValidation function takes a dataset and runs CrossValidation "iter"
+  number of times and then return the average training and validation error.
+  It shuffles the dataset in every iteration.
+
+*/
+
+Error AvgCrossValidation(arma::mat& dataset, size_t numLabels, size_t iter,
+                         const size_t hiddenLayerSize, const size_t maxEpochs)
+{
+
+/*
+  @params dataset The dataset inclusive of the labels. Assuming the last 
+                  "numLabels" number of rows are the labels which the model
+                  has to predict. 
+  @params numLabels number of rows which are the output labels in the dataset.
+
+  @params iter The number of times Cross Validation has to be run.
+
+  @params hiddenLayerSize The number of nodes in the hidden layer.
+
+  @params maxEpochs The maximum number of epochs for the training.
+
+  @params avgError Returns the train and validation error averaged over "iter"
+                   number of Cross Validation results.
+
+*/
+  Error avgError = {0.0, 0.0};
+
+  for (size_t i = 0; i < iter;)
+  {
+    dataset = arma::shuffle(dataset, 1);
+
+    arma::mat trainData = dataset.submat(0, 0, dataset.n_rows - 1 - numLabels, 
+                                         dataset.n_cols - 1);
+    arma::mat trainLabels = dataset.submat(dataset.n_rows - numLabels, 0, 
+                                           dataset.n_rows - 1, 
+                                           dataset.n_cols - 1); 
+    Error e = CrossValidation(trainData, trainLabels, 10, hiddenLayerSize, 
+                              maxEpochs);
+
+    // The check is to prevent nan values from affecting the result.
+    if (arma::is_finite(e.trainError) && arma::is_finite(e.validationError))
+    {
+      avgError.trainError += e.trainError;
+      avgError.validationError += e.validationError;
+      ++i;
+    }
+    
+  }
+
+  avgError.trainError /= iter;
+  avgError.validationError /= iter;
+
+  return avgError;
+}
+
+
+/*Test case for the Iris Dataset */
+
+BOOST_AUTO_TEST_CASE(IrisDataset)
+{
+  arma::arma_rng::set_seed_random();
```

Comment:
We set the random seed for each build. Enabling us to reuse the seed if something breaks to e.g. track down an error that only occurs in certain situations. If you reset the seed for the test case, we can't do the same for this test suite.